nengo_gui/sim.py

```diff
@@ -0,0 +1,436 @@
+
+        self.create_components()
+
+    def create_components(self):
+        """Generate the actual Components from the Templates"""
+        self.components = []
+        self.template_uids = {}
+        for k, v in self.locals.items():
+            if isinstance(v, nengo_gui.components.component.Template):
+                self.template_uids[v] = k
+                c = v.create(self)
+                self.sim_server.component_uids[c.uid] = c
+                self.components.append(c)
+
+        # this ensures NetGraph, AceEditor, and SimControl are first
+        self.components.sort(key=lambda x: x.component_order)
+
+    def add_template(self, template):
+        """Add a new Component to an existing Sim."""
+        c = template.create(self)
+        self.sim_server.component_uids[c.uid] = c
+
+        self.components.append(c)
+
+        return c
+
+    def execute(self, code):
+        """Run the given code to generate self.model and self.locals.
+
+        The code will be stored in self.code, any output to stdout will
+        be a string as self.stdout, and any error will be in self.error.
+        """
+        locals = {}
+        locals['nengo_gui'] = nengo_gui
+        locals['__file__'] = self.filename
+
+        self.code = code
+        self.error = None
+        self.stdout = ''
+
+        patch = nengo_gui.monkey.Patch(self.filename)
+        try:
+            with patch:
+                exec(code, locals)
+        except nengo_gui.monkey.StartedSimulatorException:
+            # no running a simulator inside a script
+            pass
+        except nengo_gui.monkey.StartedVizException:
+            # no running nengo_gui inside a script
+            pass
+        except:
+            line = nengo_gui.monkey.determine_line_number()
+            self.error = dict(trace=traceback.format_exc(), line=line)
+        self.stdout = patch.stdout.getvalue()
+
+        # make sure we've defined a nengo.Network
+        model = locals.get('model', None)
+        if not isinstance(model, nengo.Network):
+            if self.error is None:
+                line = len(code.split('\n'))
+                self.error = dict(trace='must declare a nengo.Network '
+                                  'called "model"', line=line)
+            model = None
+
+        self.model = model
+        self.locals = locals
+
+    def load_config(self):
+        """Load the .cfg file"""
+        config = nengo_gui.config.Config()
+        self.locals['nengo_gui'] = nengo_gui
+        self.locals['_viz_config'] = config
+        fname = self.filename_cfg
+        if os.path.exists(fname):
+            with open(fname) as f:
+                config_code = f.readlines()
+            for line in config_code:
+                try:
+                    exec(line, self.locals)
+                except Exception as e:
+                    if self.sim_server.interactive:
+                        logging.debug('error parsing config: %s' % line)
```

Comment:
`logging.debug('error parsing config: %s', line)` works too and is apparently preferred.